app/src/main/kotlin/com/flxrs/dankchat/preferences/userdisplay/UserDisplayAdapter.kt

```diff
@@ -0,0 +1,123 @@
+package com.flxrs.dankchat.preferences.userdisplay
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import com.flxrs.dankchat.R
+import com.flxrs.dankchat.databinding.AddItemBinding
+import com.flxrs.dankchat.databinding.UserDisplayItemBinding
+import com.flxrs.dankchat.utils.extensions.getContrastTextColor
+import com.flxrs.dankchat.utils.extensions.toARGBInt
+import com.flxrs.dankchat.utils.extensions.toHexCode
+import com.google.android.material.button.MaterialButton
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.rarepebble.colorpicker.ColorPickerView
+
+class UserDisplayAdapter(val entries: MutableList<UserDisplayItem>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```

Comment:
Don't use the MutableList approach, use the ListAdapter approach with callbacks and introduce a ViewModel that does all the changes to the ui state. This way we can directly sync it with the db

Reply:
gone with the way you suggested, much better code :+1: